src/server/req_manager.c

```diff
@@ -1766,6 +1786,191 @@ mgr_queue_unset(struct batch_request *preq)
 	reply_ack(preq);
 }
 
+/**
+ * @brief
+ *	Reset the "inventory Mom" for a vnode_pool if the specified Mom is the
+ *	current inventory Mom.  Done when she is down or deleted from the pool.
+ *
+ * @param[in] pmom - Pointer to the Mom (mominfo_t) structure of the Mom
+ *	being removed/marked down.
+ */
+void 
+reset_pool_inventory_mom(mominfo_t *pmom)
+{
+	mom_svrinfo_t *psvrmom = (mom_svrinfo_t *)(pmom->mi_data);
+	vnpool_mom_t  *ppool = vnode_pool_mom_list;
+	mominfo_t     *pxmom;
+	mom_svrinfo_t *pxsvrmom;
+	int           i;
+
+	/* If this Mom is in a vnode pool and is the inventory Mom for that */
+	/* pool remove her from that role and if another Mom in the pool and */
+	/* is up, make that Mom the new inventory Mom */
+
+	if (psvrmom->msr_vnode_pool != 0) {
```

Comment:
There are many places where we try to find whether or not a mom belongs to a vnode_pool. Can we make this a function?